### ISSUE TITLE
Stop providing configuration for deprecated formatting rules

### ISSUE THREAD

**Issue by m-akinc** [ni/javascript-styleguide#148]
Formatting rules were [deprecated in ESLint 8.53.0](https://eslint.org/blog/2023/10/deprecating-formatting-rules/). As suggested in the linked document, we should probably replace these with their counterparts from `@stylistic/eslint-plugin-js(ts)`.

**Comment by jattasNI** [comment]
Consensus among our working group is to migrate to the new `stylistic` packages in order to get off of deprecated rules. But we have until ESLint v10 to do so since that's when the deprecated rules will be deleted.

Longer term we could look at using one of the formatters in the linked blog post instead. Probably not prettier since it isn't very fast or configurable, but the [dprint](https://dprint.dev/) one looks worth exploring.

We don't anticipate major client impact. If any clients were overriding any of the now deprecated rules, they may need to do a bit of work to uptake this migration. We should inspect a couple client projects to see if that's the case, but it doesn't seem likely.

This is available to work on when someone has bandwidth.

**Comment by jattasNI** [comment]
I scanned Skyline for usages of the deprecated rules. The only instances I found were:

- two projects configure `eol-last: error`
- one inline disable of `function-paren-newline`
- one inline disable and one project-wide configuration of `indent`
- one project disables `lines-between-class-members`
- one project unnecesarily sets `no-trailing-spaces: error`
- a few inline disables of `object-curly-newline`
- one inline disable of quote-props
- one override of `spaced-comment`

In summary: nothing that should stop us from doing this work but a handful of places that might need to be looked at when uptaking this change if the rule names or configurations change.
